Unittests for cloud_pubsub output fail intermittently

### Relevant telegraf.conf:

### System info:

Linux
master 

### Steps to reproduce:

1. `go test -c -short ./plugins/outputs/cloud_pubsub/...`
2. `for x in {0..99}; do ./cloud_pubsub.test; done`

### Expected behavior:

All pass

### Actual behavior:

Sometimes a test will fail, here are the unique ones I see:
```
--- FAIL: TestPubSub_WriteOverCountThreshold (0.00s)
    pubsub_test.go:98: 
                Error Trace:    pubsub_test.go:98
                Error:          Not equal: 
                                expected: 2
                                actual  : 3
                Test:           TestPubSub_WriteOverCountThreshold
                Messages:       unexpected bundle count
FAIL
--- FAIL: TestPubSub_WriteMultiple (0.00s)
    pubsub_test.go:74: 
                Error Trace:    pubsub_test.go:74
                Error:          Not equal: 
                                expected: 1
                                actual  : 2
                Test:           TestPubSub_WriteMultiple
                Messages:       unexpected bundle count
FAIL
--- FAIL: TestPubSub_WriteSingle (0.00s)
    pubsub_test.go:149: expected metric to get published (value: value_1)
```

### Additional info:

[Include gist of relevant config, logs, etc.]

--- a/plugins/outputs/cloud_pubsub/pubsub.go
+++ b/plugins/outputs/cloud_pubsub/pubsub.go
@@ -1,16 +1,17 @@
 package cloud_pubsub
 
 import (
-	"cloud.google.com/go/pubsub"
 	"context"
 	"fmt"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers"
 	"golang.org/x/oauth2/google"
 	"google.golang.org/api/option"
-	"sync"
 )
 
 const sampleConfig = `
@@ -28,9 +29,9 @@ const sampleConfig = `
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
   data_format = "influx"
 
-  ## Optional. Filepath for GCP credentials JSON file to authorize calls to 
-  ## PubSub APIs. If not set explicitly, Telegraf will attempt to use 
-  ## Application Default Credentials, which is preferred. 
+  ## Optional. Filepath for GCP credentials JSON file to authorize calls to
+  ## PubSub APIs. If not set explicitly, Telegraf will attempt to use
+  ## Application Default Credentials, which is preferred.
   # credentials_file = "path/to/my/creds.json"
 
   ## Optional. If true, will send all metrics per write in one PubSub message.
@@ -55,7 +56,7 @@ const sampleConfig = `
 
   ## Optional. Specifies a timeout for requests to the PubSub API.
   # publish_timeout = "30s"
-  
+
   ## Optional. PubSub attributes to add to metrics.
   # [[inputs.pubsub.attributes]]
   #   my_attr = "tag_value"

--- a/plugins/outputs/cloud_pubsub/pubsub_test.go
+++ b/plugins/outputs/cloud_pubsub/pubsub_test.go
@@ -1,12 +1,13 @@
 package cloud_pubsub
 
 import (
+	"testing"
+
 	"cloud.google.com/go/pubsub"
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/parsers"
 	"github.com/influxdata/telegraf/testutil"
 	"github.com/stretchr/testify/assert"
-	"testing"
 )
 
 func TestPubSub_WriteSingle(t *testing.T) {

--- a/plugins/outputs/cloud_pubsub/topic_stubbed.go
+++ b/plugins/outputs/cloud_pubsub/topic_stubbed.go
@@ -1,18 +1,20 @@
 package cloud_pubsub
 
 import (
-	"cloud.google.com/go/pubsub"
 	"context"
 	"errors"
 	"fmt"
+	"runtime"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/pubsub"
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/parsers"
 	"github.com/influxdata/telegraf/plugins/serializers"
 	"google.golang.org/api/support/bundler"
-	"runtime"
-	"sync"
-	"testing"
 )
 
 const (
@@ -138,7 +140,7 @@ func (t *stubTopic) SetPublishSettings(settings pubsub.PublishSettings) {
 
 func (t *stubTopic) initBundler() *stubTopic {
 	t.bundler = bundler.NewBundler(&bundledMsg{}, t.sendBundle())
-	t.bundler.DelayThreshold = t.Settings.DelayThreshold
+	t.bundler.DelayThreshold = 10 * time.Second
 	t.bundler.BundleCountThreshold = t.Settings.CountThreshold
 	if t.bundler.BundleCountThreshold > pubsub.MaxPublishRequestCount {
 		t.bundler.BundleCountThreshold = pubsub.MaxPublishRequestCount
@@ -159,13 +161,14 @@ func (t *stubTopic) sendBundle() func(items interface{}) {
 
 		for _, msg := range bundled {
 			r := msg.stubResult
+			for _, id := range r.metricIds {
+				t.published[id] = msg.Message
+			}
+
 			if r.sendError {
 				r.err <- errors.New(errMockFail)
 			} else {
 				r.done <- struct{}{}
-			}
-			for _, id := range r.metricIds {
-				t.published[id] = msg.Message
 			}
 		}
 

